geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZAddIntegrationTest.java

```diff
@@ -0,0 +1,223 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.JedisCluster;
+import redis.clients.jedis.Protocol;
+import redis.clients.jedis.params.ZAddParams;
+
+import org.apache.geode.redis.RedisIntegrationTest;
+import org.apache.geode.redis.internal.RedisConstants;
+import org.apache.geode.test.awaitility.GeodeAwaitility;
+
+public abstract class AbstractZAddIntegrationTest implements RedisIntegrationTest {
+  private JedisCluster jedis;
+  private static final int REDIS_CLIENT_TIMEOUT =
+      Math.toIntExact(GeodeAwaitility.getTimeout().toMillis());
+  private static final String SORTED_SET_KEY = "ss_key";
+  private static final int INITIAL_MEMBER_COUNT = 5;
+
+  @Before
+  public void setUp() {
+    jedis = new JedisCluster(new HostAndPort("localhost", getPort()), REDIS_CLIENT_TIMEOUT);
+  }
+
+  @After
+  public void tearDown() {
+    flushAll();
+    jedis.close();
+  }
+
+  @Test
+  public void zaddErrors_givenWrongKeyType() {
+    final String STRING_KEY = "stringKey";
+    jedis.set(STRING_KEY, "value");
+    assertThatThrownBy(
+        () -> jedis.sendCommand(STRING_KEY, Protocol.Command.ZADD, STRING_KEY, "1", "member"))
+            .hasMessage("WRONGTYPE " + RedisConstants.ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void zaddErrors_givenTooFewArguments() {
+    assertAtLeastNArgs(jedis, Protocol.Command.ZADD, 3);
+  }
+
+  @Test
+  public void zaddErrors_givenUnevenPairsOfArguments() {
+    assertThatThrownBy(
+        () -> jedis.sendCommand("fakeKey", Protocol.Command.ZADD, "fakeKey", "1", "member", "2"))
+            .hasMessageContaining("ERR syntax error");
+  }
+
+  @Test
+  public void zaddErrors_givenNonNumericScore() {
+    assertThatThrownBy(
+        () -> jedis.sendCommand("fakeKey", Protocol.Command.ZADD, "fakeKey", "xlerb", "member"))
+            .hasMessageContaining(ERROR_NOT_A_VALID_FLOAT);
+  }
+
+  @Test
+  public void zaddErrors_givenBothNXAndXXOptions() {
+    assertThatThrownBy(
+        () -> jedis.sendCommand("fakeKey", Protocol.Command.ZADD, "fakeKey", "NX", "XX", "1.0",
+            "fakeMember"))
+                .hasMessageContaining(ERROR_INVALID_ZADD_OPTION_NX_XX);
+  }
+
+  @Test
+  public void zaddStoresScores_givenCorrectArguments() {
+    Map<String, Double> map = makeMemberScoreMap("member_", INITIAL_MEMBER_COUNT, 0);
+
+    Set<String> keys = map.keySet();
+    Long count = 0L;
+
+    for (String member : keys) {
+      Double score = map.get(member);
+      Long res = jedis.zadd(SORTED_SET_KEY, score, member);
+      Assertions.assertThat(res).isEqualTo(1);
+      Assertions.assertThat(jedis.zscore(SORTED_SET_KEY, member)).isEqualTo(score);
```

Comment:
The `Assertions` qualifier is not needed here, since there's already a static import for `assertThat()` in this class.

Reply:
Thought I'd cleaned all those up. Gone.